### PR TITLE
camlp5: improve camlp5 formula test

### PR DESCRIPTION
**Pull request by DanGrayson**
... by ensuring that the ocaml compiler provided by the package ocaml is up to
date enough.  We do that by adding a random compiled file from the ocaml
package ( bigarray.cma ) to the command line of the current test,
so camlp5 will try to link it in with the others.  If that compiled file is too
old, then the magic number stored in it will be smaller than the magic number
in the compiled files used by camlp5.

There is a discussion about this issue here:

  https://discourse.brew.sh/t/prebuilt-bottle-dependencies/3853

In addition to the tests below, I have verified that `brew test camlp5` will now fail if the current
pre-built camlp5 bottle is installed.

- [ yes] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [yes ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [yes ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ yes] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ yes] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by DanGrayson** [comment]
Maybe I should add a comment to the effect that an error message like
```
Error while loading "/usr/local/opt/ocaml/lib/ocaml/bigarray.cma": interface mismatch on Stdlib__complex.
```
is what one expects when the ocaml package is too new, and that a dissimilar error could be
due to the test becoming broken.

**Comment by DanGrayson** [comment]
If the test for `coq` had included building it from sources, it would have failed, because of the old `camlp5` package:
```
brew$ brew install --build-from-source coq
brew install --build-from-source coq
==> Downloading https://github.com/coq/coq/archive/V8.8.2.tar.gz
Already downloaded: /Users/brew/Library/Caches/Homebrew/downloads/a4948cdfc5134f2819ef208fe8ec948d4f0f5d87b20bead0348eebab8895de65--coq-8.8.2.tar.gz
==> ./configure -prefix /usr/local/Cellar/coq/8.8.2 -mandir /usr/local/Cellar/coq/8.8.2/share/man -emacslib /us
==> make world
Last 15 lines from /Users/brew/Library/Logs/Homebrew/coq/02.make:
OCAMLC -c grammar/q_util.mli
244 states, 858 transitions, table size 4896 bytes
314 states, 4454 transitions, table size 19700 bytes
2927 additional bytes used for bindings
189 states, 503 transitions, table size 3146 bytes
CAMLP5O   ide/coqide_main.ml4
OCAMLDEP  checker/MLFILES checker/MLIFILES
File "grammar/q_util.mli", line 1:
Error: /usr/local/Cellar/camlp5/7.07/lib/ocaml/camlp5/mLast.cmi
is not a compiled interface for this version of OCaml.
It seems to be for an older version of OCaml.
make[1]: *** [grammar/q_util.cmi] Error 2
make[1]: *** Waiting for unfinished jobs....
2654 states, 8247 transitions, table size 48912 bytes
make: *** [submake] Error 2

READ THIS: https://docs.brew.sh/Troubleshooting
```

File: Formula/camlp5.rb [hunk 1]
```diff
@@ -21,9 +21,11 @@ def install
     (lib/"ocaml/camlp5").install "etc/META"
   end
 
+  ocaml = Formula["ocaml"]
```
Comment (MikeMcQuaid) [inline Formula/camlp5.rb]: Put this inside the `test do` instead, thanks 👍

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by MikeMcQuaid** [comment]
Great work here @DanGrayson, thanks!

> In addition to the tests below, I have verified that `brew test camlp5` will now fail if the current
> pre-built camlp5 bottle is installed.

Could you add a `revision 1` to the formula which will ensure that when we pull this we also pull an updated `camlp5` bottle?



> Maybe I should add a comment to the effect that an error message like

Yeh a (Ruby) inline comment would be great 👍 

Thanks again!

**Comment by DanGrayson** [comment]
Okay, done.

I assume something will automatically update the hashcodes of the bottles.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Thanks so much for your first contribution! Without people like you submitting PRs we couldn't run this project. You rock, @DanGrayson!

**Comment by DanGrayson** [comment]
You're welcome!  Homebrew is well done, so it's worth getting it perfect.

**Pull request closed without merging**